### PR TITLE
3d Demo: Try to read bmp files

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Add the possibility to read a directory with bmp files which are together a voxel image

### CODE REVIEW

**Comment by afabri** [comment]
@lrineau do you know how not to apply the Gaussian Smoothing [here](https://github.com/CGAL/cgal/blob/master/Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Io_image_plugin.cpp#L1327)?

**Comment by lrineau** [comment]
> @lrineau do you know how not to apply the Gaussian Smoothing [here](https://github.com/CGAL/cgal/blob/master/Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Io_image_plugin.cpp?rgh-link-date=2022-11-25T09%3A20%3A10Z#L1327)?

You can skip the smoother, and do:
```c++
auto vtk_image = dicom_reader->GetOutput();
```

**Comment by afabri** [comment]
I had tried that but it is not enough.  It additionally needs a call to `Update()` and then it works. The smoothing should probably be a choice to tick in the demo,

File: CGAL_ImageIO/include/CGAL/IO/read_vtk_image_data.h [hunk 1]
```diff
@@ -69,26 +69,55 @@ read_vtk_image_data(vtkImageData* vtk_image, Image_3::Own owning = Image_3::OWN_
...
+    image->data = ::ImageIO_alloc(dims_n * image->wdim);
+    std::cerr << "GetNumberOfTuples() = " << vtk_image->GetPointData()->GetScalars()->GetNumberOfTuples() << "\n"
+              << "components = " << cn << "\n"
+              << "wdim = " << image->wdim << "\n"
+              << "image->size() = " << dims_n << std::endl;
+
+    if(cn == 1) {
+      vtk_image->GetPointData()->GetScalars()->ExportToVoidPointer(image->data);
+    } else {
+      std::cerr << "Warning: input has " << cn << " components; only the value of the first component will be used." << std::endl;
```
Comment (afabri) [inline CGAL_ImageIO/include/CGAL/IO/read_vtk_image_data.h:105]: Can we check that the components have the same value (in order to encode a grey value)

File: CGAL_ImageIO/include/CGAL/IO/read_vtk_image_data.h [hunk 2]
```diff
@@ -69,26 +69,55 @@ read_vtk_image_data(vtkImageData* vtk_image, Image_3::Own owning = Image_3::OWN_
...
+      // cast the data void pointers to make it possible to do pointer arithmetic
+      char* src = static_cast<char*>(vtk_image->GetPointData()->GetScalars()->GetVoidPointer(0));
+      char* dest = static_cast<char*>(image->data);
+
+      for(int i=0; i<dims_n; ++i)
+      {
+        // multiply by image->wdim because we casted to char* and not the actual data type
+        memcpy(dest + image->wdim*i, src + cn*image->wdim*i, image->wdim * sizeof(char));
+
+        // Check that we are not discarding useful data
```
[the remark this answers is not in the document]
Comment (MaelRL) [reply CGAL_ImageIO/include/CGAL/IO/read_vtk_image_data.h]: Yes, that's the point of these two assertions: `cn` is the number of components, and it checks that the second and third components (possibly out of more than 3) are identical to the first one.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-133

**Pull request merged**